### PR TITLE
fix(DB/SAI): Make generically spawned traps invisible

### PR DESCRIPTION
**Pull request by Eronox**
## Changes Proposed:
- Set all gameobjects with displayid 327 and type 6 (trap) to invisible by setting data11 column to 1. 

This should fix all the generically spawned traps.

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/8973
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/9496
- Closes chromiecraft/chromiecraft#2469

## SOURCE:

https://www.wowhead.com/object=128308/shallow-grave#comments

## Tests Performed:

- SQL tested
> /* Affected rows: 18  Found rows: 0  Warnings: 0  Duration for 3 queries: 0,015 sec. */

**This PR fixes several traps, this is how to test the ZF graves traps:**

- **Tested in game**: Making sure crystal is invisible
- **Tested in game**: Making sure traps still works as intended _(spawn adds)_
- **Tested in game**: Making sure gameobject despawns  _(on `.gm off`)_
- **Tested in game**: Making sure trap cannot be triggered twice   _(on `.gm off`)_

## How to Test the Changes:

1. `.tele zulfarrak`
2. Go to the graves
3. ` .gm off`
4. Try loot some graves. _Note_: _Not all graves has the traps on it, as intended_

## Known Issues and TODO List:

The despawn timer is not verified but sources says between 5-10minutes. 
Right now its ON 5 minutes after first loot, but it should probably be changed to a BETWEEN event and a real source.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by Nyeriah — APPROVED** [review]

**Comment by Eronox** [comment]
Actually, this issue seems to be deeper than I thought. 

Traps have stealth flag but they still appear, just like if you walk to a rogue in stealth, and this gameobject just didnt have any displayid to show when being near. 

The issue seems currently to be that it should be invisible instead of stealthed

**Comment by Shineslippers** [comment]
@Eronox do i have to create one report for Resonite Crystal objects 178104, 178105, 178106 or this PR will fix it?
WoWScrnShot_120521_120345

**Comment by Nyeriah** [comment]
From the talks we had yesterday on Discord, the only pattern I can see in this is that absolutelly all those traps that spawn creatures use that display ID.

After researching, I found out that traps are serverside objects that cannot be sniffed. I don't know Trinity got the values we got at the moment.

We could make all traps with that displayID invisible (in that part of the code you showed me yesterday), I think it works for me alright.

**Comment by patou01** [comment]
Just a quick comment as I think I recognize part of an issue I saw before, it could be that just have a shit ton of unnecessary gameobjects.  The original report suggests changing gameobject 128403 fixes it, but you don't touch it here, is that gameobject not relevant?

**Comment by Nyeriah** [comment]
> Just a quick comment as I think I recognize part of an issue I saw before, it could be that just have a shit ton of unnecessary gameobjects. The original report suggests changing gameobject 128403 fixes it, but you don't touch it here, is that gameobject not relevant?

Those are not unnecesarry gameobjects. They are the traps spawned by the gameobject_template (sniffed, wdb data), and the traps spawn the creatures. Changing gameobject_template 128403 is a hack and should not be done, reason why I did not consider it.

**Comment by Eronox** [comment]
After some investigation and help from lovely Ms. @Nyeriah, we found out that it's the data11 field that handles the invisibility switch. 

Updating PR to use data11 instead of stealth field, data9. 
Also adding fix for https://github.com/azerothcore/azerothcore-wotlk/issues/9496 at the same time

**Comment by Eronox** [comment]
> From the talks we had yesterday on Discord, the only pattern I can see in this is that all of those traps that spawn creatures use that display ID.
> 
> After researching, I found out that traps are serverside objects that cannot be sniffed. I don't know how Trinity got the values we got at the moment.
> 
> We could make all traps with that displayID invisible (in that part of the code you showed me yesterday), I think it works for me alright.

Or maybe a generic SQL update to all traps using the displayid, setting the found data11 to 1 having them invisible, instead of forcing it through core. What you say?

**Comment by Nyeriah** [comment]
> > From the talks we had yesterday on Discord, the only pattern I can see in this is that all of those traps that spawn creatures use that display ID.
> > After researching, I found out that traps are serverside objects that cannot be sniffed. I don't know how Trinity got the values we got at the moment.
> > We could make all traps with that displayID invisible (in that part of the code you showed me yesterday), I think it works for me alright.
> 
> Or maybe a generic SQL update to all traps using the displayid, setting the found data11 to 1 having them invisible, instead of forcing it through core. What you say?

Yes, peachy, go on :)

**Comment by Eronox** [comment]
> @Eronox do i have to create one more report for Resonite Crystal objects 178104, 178105, 178106 or this PR will fix it?

If we do this this manual way, which might be safer, then the addition is to just the the linked trap of resonite crystal invisible by setting data11 to 1. 

I've added it to the PR :)

**Comment by Eronox** [comment]
All of these traps should now also be invisible. 😄 
I've tested 4 of them by going ingame.

Note: `.gm off` or they show

image

File: data/sql/updates/pending_db_world/rev_1638658608.sql [hunk 1]
```diff
@@ -0,0 +1,10 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1638658608');
+
+/* Zul'Farrak Shallow Grave - On Gossip Hello - Despawn after 5min */
+DELETE FROM `smart_scripts` WHERE `entryorguid` = 128308 AND `id` = 1 AND `source_type` = 1;
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(128308, 1, 1, 0, 64, 0, 100, 0, 0, 0, 0, 0, 0, 41, 300000, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'Shallow Grave - On Gossip Hello - Despawn');
+
+/* Make generically spawned traps invisible - They share displayId */
+/* For example, fixes: ZF Grave traps, shellfish traps, resonite crystal traps and more..*/
+UPDATE gameobject_template SET `data11` = 1 WHERE displayid = 327 AND TYPE = 6;
```
Comment (Nyeriah) [inline data/sql/updates/pending_db_world/rev_1638658608.sql:10]:
```suggestion
UPDATE `gameobject_template` SET `data11` = 1 WHERE `displayid` = 327 AND `type` = 6;
```

**Pull request merged**